sway/tree/view.c

```diff
@@ -625,6 +624,8 @@ void view_map(struct sway_view *view, struct wlr_surface *wlr_surface,
 	if (should_focus(view)) {
 		input_manager_set_focus(&view->container->node);
 	}
+
+	view_execute_criteria(view);
```

Comment:
The `should_focus` check needs to be done after the view has run its criteria. Otherwise, criteria might move the view to another workspace which would cause it to steal focus.